saml/src/main/java/com/linecorp/armeria/server/saml/SamlAssertionConsumerFunction.java

```diff
@@ -283,6 +287,50 @@ private Assertion getValidatedAssertion(SamlBindingProtocol bindingProtocol,
         throw new InvalidSamlRequestException("no subject found from the assertions");
     }
 
+    @VisibleForTesting
+    static boolean isSameUri(String uriString1, String uriString2) {
+        if (uriString1.equals(uriString2)) {
+            return true;
+        }
+
+        final URI uri1 = parseUri(uriString1);
+        final URI uri2 = parseUri(uriString2);
+
+        return Objects.equals(uri1.getScheme(), uri2.getScheme()) &&
+               Objects.equals(uri1.getHost(), uri2.getHost()) &&
+               Objects.equals(uri1.getPath(), uri2.getPath()) &&
+               getPortOrDefault(uri1) == getPortOrDefault(uri2);
+    }
+
+    private static URI parseUri(String uriString) {
+        try {
+            return new URI(uriString);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid URI: " + uriString, e);
+
+        }
+    }
```

Comment:
Would it make sense to use `RequestTarget.forClient()` here instead? 🤔

Reply:
Thanks, addressed. 😉